stdlib/LinearAlgebra/src/eigen.jl

```diff
@@ -275,6 +275,18 @@ function eigen(A::AbstractMatrix{T}; permute::Bool=true, scale::Bool=true, sortb
     isdiag(AA) && return eigen(Diagonal(AA); permute=permute, scale=scale, sortby=sortby)
     return eigen!(AA; permute=permute, scale=scale, sortby=sortby, jvl=jvl, jvr=jvr, jce=jce, jcv=jcv)
 end
+function eigen(A::AbstractMatrix{Float16}; permute::Bool=true, scale::Bool=true, sortby::Union{Function,Nothing}=eigsortby)
+    AA = copy_oftype(A, eigtype(Float16))
+    isdiag(AA) && return eigen(Diagonal(AA); permute=permute, scale=scale, sortby=sortby)
+    A = eigen!(AA; permute=permute, scale=scale, sortby=sortby)
+    if isreal(A.values) == false
+        Eigen(convert(Vector{Complex{Float16}}, A.values),
+            convert(AbstractArray{Complex{Float16}}, A.vectors))
+    else
+        Eigen(convert(Vector{Float16}, A.values),
+            convert(AbstractArray{Float16}, A.vectors))
+    end
```

Comment:
```suggestion
    if isreal(A.values)
        Eigen(convert(Vector{Float16}, A.values),
            convert(AbstractArray{Float16}, A.vectors))
    else
        Eigen(convert(Vector{Complex{Float16}}, A.values),
            convert(AbstractArray{Complex{Float16}}, A.vectors))
    end
```
Also, `Eigen` seems to have a few more fields actually, which potentially need to be included in the conversion or at least listed here in the constructor. It may be worth actually adding such a type converter as we have for SVD and Cholesky.

Reply:
What fields are you talking about? Would it be better to open a new PR on the converter since this PR is already cluttered? 😅